### PR TITLE
Rebuilds caching

### PR DESCRIPTION
**Pull request by shcheklein**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please chose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by rogermparent** [comment]
I got this in a working state primarily by pinning three Gatsby plugins to earlier versions that don't use the Gatsby Jobs V2 API:

```
"gatsby-plugin-manifest": "2.2.23",
"gatsby-plugin-sharp": "2.2.32",
"gatsby-transformer-sharp": "2.2.23",
```
For some reason, this new API is overeager to invalidate its cache as evidenced by [this issue](https://github.com/gatsbyjs/gatsby/issues/23365). My local machine builds and Gatsby Cloud get around this somehow (maybe [this new beta plugin from Netlify](https://github.com/jlengstorf/netlify-plugin-gatsby-cache) as well, but I haven't tested it yet). More research and needs to be done to truly fix this cache issue going into the future, but this PR's current state serves as a fix for until then.

There are also lots of refactors I ended up doing in between tests, which can (and almost certainly will) be changed. A big one is moving most S3 logic out of the deploy script and into `s3-utils`.

File: package.json [hunk 1]
```diff
@@ -164,8 +164,6 @@
     ]
   },
   "cacheDirectories": [
-    "node_modules",
-    ".cache",
-    "public"
+    "node_modules"
```
Comment (shcheklein) [inline package.json:167]: I think Heroku does this by default (cache node_modules)

do we actually want to cache `.cache` and `public` (+ then do sync with S3) - would it make it faster? ( I would imagine it is the same S3)

Comment (rogermparent) [reply package.json:167]: I believe caching `.cache` and `public` on Heroku adds them to the slug size, and is part of why this repo was starting to push the limit and we started with S3 in the first place.

**Comment by shcheklein** [comment]
@rogermparent LGTM!

Some minor comment about caching dirs on Heroku.

Also, let's bump v1 to v2 in the circle ci config to overcome the issue with dependencies and pass the check.

**Pull request merged**